apps/judicial-system/web/src/routes/Prosecutor/CreateDetentionRequest/StepTwo/StepTwo.tsx

```diff
@@ -522,40 +537,49 @@ export const StepTwo: React.FC = () => {
                             const cRestrictions =
                               copyOfState.custodyRestrictions
 
-                            restrictions.splice(
-                              restrictions.indexOf(
-                                target.value as CaseCustodyRestrictions,
-                              ),
-                              1,
-                            )
+                            if (restrictions) {
+                              restrictions.splice(
+                                restrictions.indexOf(
+                                  target.value as CaseCustodyRestrictions,
+                                ),
+                                1,
+                              )
 
-                            cRestrictions.splice(
-                              restrictions.indexOf(
-                                target.value as CaseCustodyRestrictions,
-                              ),
-                              1,
-                            )
+                              if (cRestrictions) {
+                                cRestrictions.splice(
+                                  restrictions.indexOf(
+                                    target.value as CaseCustodyRestrictions,
+                                  ),
+                                  1,
+                                )
+                              }
+                            }
                           }
 
                           // Set the updated state as the state
                           setWorkingCase(copyOfState)
 
                           // Save case
-                          await updateCase(
-                            workingCase.id,
-                            parseArray(
-                              'requestedCustodyRestrictions',
-                              copyOfState.requestedCustodyRestrictions,
-                            ),
-                          )
+                          if (copyOfState.requestedCustodyRestrictions) {
+                            await updateCase(
+                              workingCase.id,
+                              parseArray(
+                                'requestedCustodyRestrictions',
+                                copyOfState.requestedCustodyRestrictions,
+                              ),
+                            )
+                          }
+
                           // TODO: COMBINE IN A SINGLE API CALL
-                          await updateCase(
-                            workingCase.id,
-                            parseArray(
-                              'custodyRestrictions',
-                              copyOfState.custodyRestrictions,
-                            ),
-                          )
+                          if (copyOfState.custodyRestrictions) {
```

Comment:
Consider copying the value in the judge flow. By copying this way, any values set by the judge will be overwritten here. Or am I missing something?